src/components/core/GeoChart.tsx

```diff
@@ -40,28 +52,49 @@ export type IGeoChartInnerProps = ICoreGeoChartProps &
     IDataSourceProviderInjectedProps &
     WrappedComponentProps;
 
+export interface IGeoChartInnerState {
+    legendIndexEnabled: boolean[];
+    legendNamesEnabled: string[];
+}
+export interface IGeoChartInnerOptions {
+    geoData: IGeoData;
+    series: any;
+    sizeValues: number[];
+}
 /**
  * Geo Chart react component
  */
-export class GeoChartInner extends BaseVisualization<IGeoChartInnerProps, {}> {
+export class GeoChartInner extends BaseVisualization<IGeoChartInnerProps, IGeoChartInnerState> {
     public static defaultProps: Partial<IGeoChartInnerProps> = {
         ...commonDefaultProps,
         chartRenderer: renderChart,
         legendRenderer: renderLegend,
-        legendPostion: TOP,
+        legendPosition: TOP,
     };
+    private geoChartOptions: IGeoChartInnerOptions;
+    constructor(props: IGeoChartInnerProps) {
+        super(props);
+        this.state = {
+            legendIndexEnabled: [],
+            legendNamesEnabled: [],
+        };
+        this.onLegendItemClick = this.onLegendItemClick.bind(this);
+    }
 
     public componentDidUpdate(prevProps: IGeoChartInnerProps) {
+        console.info("update", this.props.execution); //tslint:disable-line
         if (!isEqual(this.props.execution, prevProps.execution)) {
-            this.validateData();
+            this.transformationData();
         }
     }
 
     public componentDidMount() {
-        this.validateData();
+        console.info("mount", this.props.execution); //tslint:disable-line
```

Comment:
please remove `console` calls when this PR is ready

Reply:
fixed